erpnext/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -107,8 +107,8 @@ def check_existing(self):
 		if not self.salary_slip_based_on_timesheet:
 			ret_exist = frappe.db.sql("""select name from `tabSalary Slip`
 						where start_date = %s and end_date = %s and docstatus != 2
-						and employee = %s and name != %s""",
-						(self.start_date, self.end_date, self.employee, self.name))
+						and employee = %s and name != %s and payroll_entry= %s """,
```

Comment:
the duplicate should be checked irrespective of it is created against payroll entry or not.